### PR TITLE
Add state management for banner notification

### PR DESCRIPTION
**Pull request by jonathonherbert**
## What's changed?

We'd like to display banner notifications in the app. This adds state management for banner notifications, to be used by a component in a downstream PR.

## Implementation notes

I've added duplicate-management code, as there are many things that could display this banner, but we only want one of it. YAGNI?

### General
- [x] 🤖 Relevant tests added
- [x] ✅ CI checks / tests run locally
- [ ] 🔍 Checked on CODE

### Client
- [x] 🚫 No obvious console errors on the client (i.e. React dev mode errors)
- [x] 🎛️ No regressions with existing user interactions (i.e. all existing buttons, inputs etc. work)
- [ ] 📷 Screenshots / GIFs of relevant UI changes included

### CODE REVIEW

File: fronts-client/src/bundles/notificationsBundle.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        };
+      }
+
+      return {
+        banners: [
+          ...state.banners,
+          {
+            id: action.payload.id,
+            message: action.payload.message,
+            duplicates: 1
```
Comment (jennygrahamjones) [inline fronts-client/src/bundles/notificationsBundle.ts]: If there is 1 duplicate, does that mean that only a single instance of the notification has been added?

Comment (jonathonherbert) [reply fronts-client/src/bundles/notificationsBundle.ts]: Yes, the naming here isn't quite right. Perhaps `instances`?

Comment (jennygrahamjones) [reply fronts-client/src/bundles/notificationsBundle.ts]: Perhaps stick with the name and default to `0` then. I think that would make sense!

**Comment by jonathonherbert** [comment]
Closing as #1205 contained the changes already, and I mistakenly merged it first 🙄

**Pull request closed without merging**